core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -133,19 +142,29 @@ export class Question {
     return null;
   }
 
-  getUnaddressedMisconceptionNames(misconceptionsBySkill: {}): string[] {
+  getUnaddressedMisconceptionNames(
+      misconceptionsBySkill: Record<string, Misconception[]>
+  ): string[] {
     var answerGroups = this._stateData.interaction.answerGroups;
-    var taggedSkillMisconceptionIds = {};
+    var taggedSkillMisconceptionIds: Record<string, boolean> = {};
     for (var i = 0; i < answerGroups.length; i++) {
       if (!answerGroups[i].outcome.labelledAsCorrect &&
-          answerGroups[i].taggedSkillMisconceptionId !== null) {
-        taggedSkillMisconceptionIds[
-          answerGroups[i].taggedSkillMisconceptionId] = true;
+        answerGroups[i].taggedSkillMisconceptionId !== null) {
+          type MisconceptionId = keyof typeof taggedSkillMisconceptionIds;
+          var misconceptionId = (
+            answerGroups[i].taggedSkillMisconceptionId as MisconceptionId
+          );
+          taggedSkillMisconceptionIds[misconceptionId] = true;
```

Comment:
Why are the changes here necessary?

Reply:
here `misconceptionId` may be null and we can't assign null to index type in `taggedSkillMisconceptionIds`. So we explicitly identify the type `misconceptionId` by using `keyof typeof`. 

and then we have to define `taggedSkillMisconceptionIds` because without defining the object type, it by default will be never. So when you tried to add `type MisconceptionId`,  it was a type mismatch, and so it threw the error. Hence define type of object by using `Record<string, boolean>`.

Reply:
Ah I see, thanks @JeeveshGarg!